### PR TITLE
Add support for secondary solid buttons

### PR DESCRIPTION
**Pull request by DilwoarH**
# What
Solid secondary buttons styles from GOVUK design systems

# Why
User research has shown that solid secondary buttons worked well

Screenshot 2020-12-09 at 11 09 08

### CODE REVIEW

**Comment by DilwoarH** [comment]
required by https://github.com/alphagov/collections/pull/2133

File: app/views/govuk_publishing_components/components/_button.html.erb [hunk 1]
```diff
@@ -2,6 +2,9 @@
   shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)
   local_assigns[:classes] = shared_helper.classes
   button = GovukPublishingComponents::Presenters::ButtonHelper.new(local_assigns)
+
+          #require 'pry'; binding.pry
```
Comment (alex-ju) [inline app/views/govuk_publishing_components/components/_button.html.erb:6]: we may want to remove the debugger

**Review by alex-ju — COMMENTED** [review]
Code looks good (aside from the comment on debugger), but I'm a bit wary that we're adding a new design without explaining when we should be using the solid version and not the secondary. Is there any particular reasoning provided by the designer that we can add to the guidance? Also, I'm aware @andysellick had a meeting with the Design System and GOV.UK's Head of Design – we might be able to capture that decision when adding this new version to the library.

File: app/views/govuk_publishing_components/components/_button.html.erb (hunk 1 above)
Comment (DilwoarH) [reply app/views/govuk_publishing_components/components/_button.html.erb:6]: ops

**Comment by leenagupte** [comment]
@DilwoarH @alex-ju Shouldn't the style for the secondary button be replaced completely if it's going to match the design system rather than add another option? https://design-system.service.gov.uk/components/button/

**Comment by DilwoarH** [comment]
> @DilwoarH @alex-ju Shouldn't the style for the secondary button be replaced completely if it's going to match the design system rather than add another option? https://design-system.service.gov.uk/components/button/

We got a bit of naming work to do to assign it with the Design system. The secondary button that we currently have is used within the admin interfaces so they would need to be migrated over

**Comment by alex-ju** [comment]
> @leenagupte: Shouldn't the style for the secondary button be replaced completely if it's going to match the design system rather than add another option? https://design-system.service.gov.uk/components/button/

### Context
The current designs for secondary buttons (named 'secondary' and 'secondary quiet') in Publishing Components are used in a few applications (including Content Published, Collections Publisher) and were built before there was a 'secondary' variation in the Design System. When the Design System rolled out the secondary variation, designers in GOV.UK at the time felt that we shouldn't replace them straight away, as they tested well in research and the new ones were not coming with a solution for what we call 'secondary quiet'. Currently, the Design System is looking to address this 'secondary quiet' use-case, but we don't know how long will it take and how the new proposal will be received by designers in GOV.UK. A classic case of design debt I would say.

### Options
- If we do want to override the Publishing Components' secondary style with the Design System one, we should make sure that the affected products are ok with it (the least, check this with Product Manager) as it is, in my humble opinion, a visual breaking change.
- If we decide to create a new variation, we should also think of how will we converge this new 'secondary solid' with 'secondary' in the near future.

**Review by alex-ju — APPROVED** [review]
Technically correct. I will leave the decision to @DilwoarH and @leenagupte on whether or not to proceed with this approach or replace the current 'secondary' as explained in the comment above.

**Comment by DilwoarH** [comment]
> Code looks good (aside from the comment on debugger), but I'm a bit wary that we're adding a new design without explaining when we should be using the solid version and not the secondary. Is there any particular reasoning provided by the designer that we can add to the guidance? Also, I'm aware @andysellick had a meeting with the Design System and GOV.UK's Head of Design – we might be able to capture that decision when adding this new version to the library.

> Technically correct. I will leave the decision to @DilwoarH and @leenagupte on whether or not to proceed with this approach or replace the current 'secondary' as explained in the comment above.

I think we (as in the community) need to sit together and decide how to proceed.

**Comment by alex-ju** [comment]
> I think we (as in the community) need to sit together and decide how to proceed.

Sounds like a good approach. Given it's a piece of design debt we'd probably need designers in the room too.

**Comment by leenagupte** [comment]
@alex-ju @DilwoarH I'm just happy that the change is being made in the gem. 💯

File: lib/govuk_publishing_components/presenters/button_helper.rb [hunk 2]
```diff
@@ -77,6 +79,7 @@ def css_classes
         css_classes << "govuk-button--start" if start
         css_classes << "gem-c-button--secondary" if secondary
         css_classes << "gem-c-button--secondary-quiet" if secondary_quiet
+        css_classes << "govuk-button--secondary" if secondary_solid
```
Comment (leenagupte) [inline lib/govuk_publishing_components/presenters/button_helper.rb:82]: Sorry if I'm being dense, but where is the new style being added? Won't this just add the existing green text version?

Comment (alex-ju) [reply lib/govuk_publishing_components/presenters/button_helper.rb:82]: This comes from the Design System (`govuk-frontend` via npm): https://github.com/alphagov/govuk-frontend/blob/cf5be781776d7bb24394dd9211f672588ae081f6/src/govuk/components/button/_index.scss#L172-L202

**Pull request merged**